### PR TITLE
Refactor motor

### PR DESCRIPTION
**Pull request by mcjcloud**
Refactor motor to make it consumable by other Go projects.

## Changes
* Move `internal/motor` to `pkg/motor`
* Convert `MotorNode` into an interface with default implementation

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202528620406598/1202708961904895) by [Unito](https://www.unito.io)

### CODE REVIEW

File: bind/motor-mobile/bind.go [hunk 1]
```diff
@@ -34,38 +35,67 @@ func Init(buf []byte) ([]byte, error) {
...
+	}
 }
 
 // Address returns the address of the wallet.
 func Address() string {
 	if instance == nil {
 		return ""
 	}
-	addr, err := instance.Wallet.Address()
+	addr, err := instance.GetWallet().Address()
```
Comment (joshLong145) [inline bind/motor-mobile/bind.go]: we should split this into two calls and check for nil on the wallet return

File: bind/motor-mobile/bind.go [hunk 2]
```diff
@@ -74,7 +104,7 @@ func Address() string {
 
 // Balance returns the balance of the wallet.
 func Balance() int {
-	return int(instance.Balance())
+	return int(instance.GetBalance())
```
Comment (joshLong145) [inline bind/motor-mobile/bind.go:111]: should be nil checked for the cast and return. this could panic.

File: bind/motor-mobile/bind.go [hunk 3]
```diff
@@ -94,7 +124,7 @@ func DidDoc() string {
 	if instance == nil {
 		return ""
 	}
-	buf, err := instance.DIDDocument.MarshalJSON()
+	buf, err := instance.GetDIDDocument().MarshalJSON()
```
Comment (joshLong145) [inline bind/motor-mobile/bind.go]: same as above, should be nil checked

**Review by joshLong145 — CHANGES_REQUESTED** [review]
looking good, but a few changes.

**Comment by joshLong145** [comment]
@brokecollegekidwithaclothingobsession I think this PR will cover yours which migrates the motor to `pkg` @mcjcloud could you also add the changes. to the task file which removed to no longer defined task from running for `bind:ios` which is defined in #483

File: bind/motor-mobile/bind.go (hunk 2 above)
Comment (mcjcloud) [reply bind/motor-mobile/bind.go:111]: `GetBalance` can't return nil

File: bind/motor-mobile/bind.go (hunk 1 above)
Comment (mcjcloud) [reply bind/motor-mobile/bind.go]: done

File: bind/motor-mobile/bind.go (hunk 3 above)
Comment (mcjcloud) [reply bind/motor-mobile/bind.go]: done

**Review by joshLong145 — APPROVED** [review]

**Pull request merged**